optimum/intel/ipex/modeling_base.py

```diff
@@ -472,6 +486,8 @@ def __init__(
 
         if self._is_ipex_exported:
             self._reorder_cache = _ipex_reorder_cache
+            transformers.generation.candidate_generator._crop_past_key_values = _ipex_crop_past_key_values
+            transformers.generation.utils._crop_past_key_values = _ipex_crop_past_key_values
```

Comment:
I don't think we should patch a transformers function here

Reply:
I patched the function before `generate` and un-patched it after `generate`. Please review my new changes, thx.